test/rebuild-yarnworkspace.ts

```diff
@@ -0,0 +1,166 @@
+        arch: process.arch,
+        projectRootPath
+      },
+      name: 'options object'
+    }
+  ];
+
+  for (const options of optionSets) {
+    describe(`core behavior -- ${options.name}`, function() {
+      this.timeout(2 * 60 * 1000);
+
+      before(resetTestModule);
+
+      before(async () => {
+        let args: any = options.args;
+        await (<any>rebuild)(args);
+      });
+
+      it('should have rebuilt top level prod dependencies', async () => {
+        const forgeMeta = path.resolve(
+          testModulePath,
+          'node_modules',
+          'ref',
+          'build',
+          'Release',
+          '.forge-meta'
+        );
+        expect(
+          await fs.pathExists(forgeMeta),
+          'ref build meta should exist'
+        ).to.equal(true);
+      });
+
+      it('should not have rebuild top level prod dependencies that are prebuilt', async () => {
+        const forgeMeta = path.resolve(
+          testModulePath,
+          'node_modules',
+          'sodium-native',
+          'build',
+          'Release',
+          '.forge-meta'
+        );
+        expect(
+          await fs.pathExists(forgeMeta),
+          'ref build meta should exist'
+        ).to.equal(false);
+      });
+
+      it('should have rebuilt children of top level prod dependencies', async () => {
+        const forgeMetaGoodNPM = path.resolve(
+          testModulePath,
+          'node_modules',
+          'microtime',
+          'build',
+          'Release',
+          '.forge-meta'
+        );
+        const forgeMetaBadNPM = path.resolve(
+          testModulePath,
+          'node_modules',
+          'benchr',
+          'node_modules',
+          'microtime',
+          'build',
+          'Release',
+          '.forge-meta'
+        );
+        expect(
+          (await fs.pathExists(forgeMetaGoodNPM)) ||
+            (await fs.pathExists(forgeMetaBadNPM)),
+          'microtime build meta should exist'
+        ).to.equal(true);
+      });
+
+      it('should have rebuilt children of scoped top level prod dependencies', async () => {
+        const forgeMeta = path.resolve(
+          testModulePath,
+          'node_modules',
+          '@newrelic/native-metrics',
+          'build',
+          'Release',
+          '.forge-meta'
+        );
+        expect(
+          await fs.pathExists(forgeMeta),
+          '@newrelic/native-metrics build meta should exist'
+        ).to.equal(true);
+      });
+
+      it('should have rebuilt top level optional dependencies', async () => {
+        const forgeMeta = path.resolve(
+          testModulePath,
+          'node_modules',
+          'zipfile',
+          'build',
+          'Release',
+          '.forge-meta'
+        );
+        expect(
+          await fs.pathExists(forgeMeta),
+          'zipfile build meta should exist'
+        ).to.equal(true);
+      });
+
+      it('should not have rebuilt top level devDependencies', async () => {
```

Comment:
I don't think the entire rebuild suite should be copied for workspaces - probably only one simple integration test and then unit tests for the new functions.